android/build.gradle

```
@@ -207,3 +207,80 @@ dependencies {
     implementation "androidx.core:core-ktx:1.6.0"
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
 }
+
+def isGHExampleApp() {
+    return project.hasProperty('isGHExampleApp') && project.property('isGHExampleApp') == "true"
+}
+
+def getAbsoluteCodegenArtifactsPaperDestination() {
+    if (!project.hasProperty('codegenArtifactsPaperDestination')) {
+        throw new Exception('[react-native-gesture-handler] Please fill codegenArtifactsPaperDestination variable in android/gradle.properties to point to the correct path to generated specs for paper')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsPaperDestination')}"
+}
+
+def getAbsoluteCodegenArtifactsSource() {
+    if (!project.hasProperty('codegenArtifactsSource')) {
+        throw new Exception('[react-native-gesture-handler] Please fill codegenArtifactsSource variable in android/gradle.properties to point to the correct path to codegenerated artifacts')
+    }
+
+    return "${project.rootDir}/../../${project.property('codegenArtifactsSource')}"
+}
+
+
+tasks.register("copyCodegenArtifacts") {
+    group 'After build tasks'
+    description 'Task which copies codegen artifacts to paper architecture'
+    
+    if (!isGHExampleApp() || !isNewArchitectureEnabled()) {
+        return
+    }
+
+    dependsOn tasks.generateCodegenArtifactsFromSchema
+
+    doLast {
+
+        def absoluteCodegenArtifactsPaperDestination = getAbsoluteCodegenArtifactsPaperDestination()
+        def absoluteCodegenArtifactsSource = getAbsoluteCodegenArtifactsSource()
+
+        def existingFiles = fileTree(absoluteCodegenArtifactsPaperDestination).matching {
+            include '**/*.java'
+        }
+
+        def generatedFiles = fileTree(absoluteCodegenArtifactsSource).matching {
+            include '**/*.java'
+        }
+
+        def existingFilesMap = [:]
+
+        existingFiles.forEach { existingFile ->
+            println existingFile
+            existingFilesMap[existingFile.name] = 1
+        }
+
+        generatedFiles.forEach { generatedFile -> 
+            if (!existingFilesMap.containsKey(generatedFile.name)) {
+                logger.warn("[react-native-gesture-handler] ${generatedFile.name} not found in paper dir, if it's used in Android you need to copy it manually and implement yourself before using auto-copy feature")
```

A small nitpick, but I believe this should be `on Android` 😄 

```suggestion
                logger.warn("[react-native-gesture-handler] ${generatedFile.name} not found in paper dir, if it's used on Android you need to copy it manually and implement yourself before using auto-copy feature.")
```

---

Changed in: https://github.com/software-mansion/react-native-gesture-handler/pull/2933/commits/7e4da8d6e5dcc10337a96a60d5e4c44c6d75283b